src/lib/address/unstoppable-domains.ts

```diff
@@ -0,0 +1,71 @@
+import { btcChainId, ethChainId } from '@shapeshiftoss/caip'
+import { Resolution } from '@unstoppabledomains/resolution'
+import { getConfig } from 'config'
+import last from 'lodash/last'
+import {
+  ResolveVanityAddress,
+  ReverseLookupVanityAddress,
+  ValidateVanityAddress,
+} from 'lib/address/address'
+import { logger } from 'lib/logger'
+
+const moduleLogger = logger.child({ namespace: ['unstoppable-domains'] })
+
+let _resolution: Resolution | undefined
+
+const getResolution = (): Resolution => {
+  const apiKey = last(getConfig().REACT_APP_ETHEREUM_NODE_URL.split('/')) ?? ''
+  if (!apiKey) moduleLogger.error('No API key found in REACT_APP_ETHEREUM_NODE_URL')
+  if (!_resolution) _resolution = Resolution.infura(apiKey)
+  return _resolution
+}
+
+// validate
+export const validateUnstoppableDomain: ValidateVanityAddress = async ({ value }) => {
+  try {
+    return getResolution().isSupportedDomain(value)
+  } catch (e) {
+    moduleLogger.trace(e, 'cannot validate unstoppable domain')
+    return false
+  }
+}
+
+const chainIdToUDTicker: Record<string, string> = {
```

Comment:
Happier with the full Unstoppable Domain terminology but I'm fine with this since it's contained in `unstoppable-domains` module, it's really just for grepping purposes

Reply:
```suggestion
const chainIdToUDTicker: Record<ChainId, string> = {
```

Also, not for this diff but we have 
https://github.com/shapeshift/web/blob/2c39be82a945c2e6504da9276008a6750d0f0d9d/src/state/slices/portfolioSlice/utils.ts#L32 (probably not living in the best place)

I'm wondering if we couldn't make things more accurate in the future, using the narrowing of `as const` to have more accurate `ChainId` typings